server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1250,9 +1249,11 @@ private void doGetRepositoryData(ActionListener<RepositoryData> listener) {
                     loaded = repositoryDataFromCachedEntry(cached);
                 } else {
                     loaded = getRepositoryData(genToLoad);
-                    // We can cache in the most recent version here without regard to the actual repository metadata version since we're
-                    // only caching the information that we just wrote and thus won't accidentally cache any information that isn't safe
-                    cacheRepositoryData(loaded, Version.CURRENT);
+                    // We can cache serialized in the most recent version here without regard to the actual repository metadata version
+                    // since we're only caching the information that we just wrote and thus won't accidentally cache any information that
+                    // isn't safe
+                    cacheRepositoryData(
```

Comment:
Still kind of awkward to right away serialize what we just read but I didn't want to make the bigger changes required for not having to serialize here in this PR + it's a rare spot to get here anyway and only applies to the first load of repository data on a master node.